test/Templater.spec.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+
+const chai = require('chai');
+const sinon = require('sinon');
+const sinonChai = require('sinon-chai');
+
+
+const Templater = require('../index').Templater;
+
+
+chai.should();
+chai.use(sinonChai);
+
+
+describe('#Templater', function () {
+
+
+    describe('#Templater.derive_language', function () {
```

Comment:
@pandafox Please check if we still need other test cases.

Reply:
The test cases looks great.

Perhaps add a test case for empty string, empty array